syntax = "proto3";

package yolo_service;

message Empty {
  // Represents an empty request or response.
}

message ImageFrame {
  bytes image_data = 1;
  int64 timestamp = 2;
}

message BoundingBox {
  float x1 = 1;
  float y1 = 2;
  float x2 = 3;
  float y2 = 4;
  int32 class_id = 5;
  float confidence = 6;
}

message PredictionBatch {
  repeated BoundingBox detections = 1;
  int64 timestamp = 2;
}

message YoloClassLabels {
  map<int32, string> class_labels = 1;
}

service YoloService {
  rpc Predict (ImageFrame) returns (PredictionBatch);
  rpc GetYoloClassLabels (Empty) returns (YoloClassLabels);
}
